lib-src/rpc2/rpc2-src/pack_helper.c

```diff
@@ -0,0 +1,402 @@
+    *old_ptr += _PAD(RPC2_KEYSIZE);
+    return 0;
+}
+
+
+int pack_int(char** new_ptr, RPC2_Integer value, char* EOB)
+{
+    if (*new_ptr + 4 > EOB)
+        return -1;
+    *(RPC2_Integer *)new_ptr = htonl(value);
+    *new_ptr += 4;
+    return 0;
+}
+
+
+int pack_unsigned(char** new_ptr, RPC2_Unsigned value, char* EOB)
+{
+    if (*new_ptr + 4 > EOB)
+        return -1;
+    *(RPC2_Unsigned *)new_ptr = htonl(value);
+    *new_ptr += 4;
+    return 0;
+}
+
+
+int pack_double(char** new_ptr, RPC2_Double value, char* EOB)
+{
+    if (*new_ptr + 8 > EOB)
+        return -1;
+    *(RPC2_Double *)new_ptr = value;
+    *new_ptr += 8;
+    return 0;
+}
+
+
+int pack_bound_bytes(char** new_ptr, char* old_ptr, char* EOB, long len)
+{
+    if (*new_ptr + len > EOB)
+        return -1;
+    memcpy(*new_ptr, old_ptr, len);
+    *new_ptr += _PAD(len);
+    return 0;
+}
+
+
+int pack_unbound_bytes(char** new_ptr, RPC2_Byte value, char* EOB)
+{
+    if (*new_ptr + 4 > EOB)
+        return -1;
+    *(RPC2_Byte *)new_ptr = value;
+    *new_ptr += 4;
+    return 0;
+}
+
+
+int pack_string(char** new_ptr, char* old_ptr)
+{
+    int length = strlen(old_ptr);
+    *(*(RPC2_Integer **)(new_ptr)) = length;
+    strcpy(*new_ptr + 4, old_ptr);
+    *(*new_ptr + 4 + length) = '\0';
+    *new_ptr += 4 + _PAD(length + 1);
+    return 0;
+}
+
+
+int pack_countedbs(char** new_ptr, char* old_ptr, RPC2_Unsigned len)
+{
+    *(*(RPC2_Unsigned **)new_ptr) = htonl(len);
+    memcpy(*new_ptr + 4, old_ptr, len);
+    *new_ptr += 4 + _PAD(len);
+    return 0;
+}
+
+
+int pack_boundedbs(char** new_ptr, char* old_ptr, RPC2_Unsigned maxLen, RPC2_Unsigned len)
+{
+    *(*(RPC2_Unsigned **)new_ptr) = htonl(maxLen);
+    if (len == 0)
+        *(RPC2_Unsigned *)(*new_ptr + 4) = 0;
+    else {
+      *(RPC2_Unsigned *)(*new_ptr + 4) = htonl(len);
+      memcpy(*new_ptr + 8, old_ptr, len);
+    }
+    *new_ptr += 8 + _PAD(len);
+    return 0;
+}
+
+
+int pack_encryptionKey(char** new_ptr, char* old_ptr, char* EOB)
+{
+    if (*new_ptr + RPC2_KEYSIZE > EOB)
+        return -1;
+    memcpy(*new_ptr, old_ptr, RPC2_KEYSIZE);
+    *new_ptr += _PAD(RPC2_KEYSIZE);
+    return 0;
+}
+
+
+int pack_struct_CallCountEntry(char** new_ptr, CallCountEntry var, char* _EOB, RPC2_Integer who)
+{
+    int rc = 0;
+    rc = pack_string(new_ptr, (char*)var.name);
+    if (rc < 0) return -1;
+    rc = pack_int(new_ptr, var.countent, _EOB);
+    if (rc < 0) return -1;
+    rc = pack_int(new_ptr, var.countexit, _EOB);
+    if (rc < 0) return -1;
+    rc = pack_int(new_ptr, var.tsec, _EOB);
+    if (rc < 0) return -1;
+    rc = pack_int(new_ptr, var.tusec, _EOB);
+    if (rc < 0) return -1;
+    rc = pack_int(new_ptr, var.counttime, _EOB);
+    if (rc < 0) return -1;
+    return 0;
+}
+
+
+int unpack_struct_CallCountEntry(CallCountEntry* new_ptr, char** old_ptr, char* _EOB, RPC2_Integer who)
+{
+    int rc = 0;
+    rc = unpack_string((unsigned char**)&((*new_ptr).name), old_ptr, _EOB, 0);
```

Comment:
What is wrong with `&(new_ptr->name)`?
Is the extra `(unsigned char**)` cast really necessary?
If so. should it not be `RPC2_String*`?